impress-org/givewp
# Feature: improve login expired message and clean up storage system

## Description

The primary goal of this is to provide a more meaningful message to users in the form builder when they got to save and their session has expired. We can later explore a UI to allow the user to log in and not lose their place, but for now we're starting with a good message.

While working on this, I also realized we have some old code for a storage system that no longer works — due to deep WP integration — and cleaned up the storage system a bit.

## Affects

The system for saving, loading, and previewing within the form builder.

## Testing Instructions

Two things to test. First, make sure that the form builder still works, especially saving and previewing forms.

Second, do the following to simulate an expired login session:
1. Open the form builder
2. Open the JS console
3. Press cmd+shift+P to open up a command console and type in "Clear site data", and run the command
4. Make some change in the form
5. Click "update" to save the form
6. An alert should pop up with the new message

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205405424043724
